File: JavaScript/ifStatements/trafficFine.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function calculateFine(speedLimit, actualSpeed) {
    const excessSpeed = actualSpeed - speedLimit;
    if (excessSpeed <= 5) {
        return 0;
    } else if (excessSpeed <= 10) {
        return 50;
    } else if (excessSpeed <= 20) {
        return 100;
    } else if (excessSpeed <= 30) {
        return 200;
    } else {
        return "You have lost your license!!";
    }
}

rl.question("Please enter the speed limit: ", (limitInput) => {
    const speedLimit = parseFloat(limitInput);

    if (isNaN(speedLimit) || speedLimit <= 0) {
        console.log("Invalid Input!");
        rl.close();
        return;
    }
    rl.question("Please enter your speed: ", (actualInput) => {
        const actualSpeed = parseFloat(actualInput);

        if (isNaN(actualSpeed) || actualSpeed < 0) {
            console.log("Invalid Input");
            rl.close();
            return;
        }

        const fine = calculateFine(speedLimit, actualInput);

        if (fine > 0) {
            console.log(`You have a speeding fine! Your total is: $${fine}`);
        } else {
            console.log(`No fine! Please continue to drive safe! `);
        }
        rl.close();
    });
});

File: ifStat/src/trafficFine.java
import java.util.Scanner;

public class trafficFine {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("What is the speed limit? ");
        double speedLimit = input.nextDouble();

        System.out.print("What was your actual speed? ");
        double actualSpeed = input.nextDouble();

        int fine;

        if(actualSpeed <= speedLimit) {
            fine = 0;
        } else if (actualSpeed <= speedLimit + 10) {
            fine = 50;
        } else if (actualSpeed <= speedLimit + 20) {
            fine = 100;
        } else {
            fine = 200;
        }
        System.out.println(fine);
    }
}